doc/en/cache.rst

```diff
@@ -0,0 +1,269 @@
+    run-last-failure: rerun last 2 failures
+    rootdir: /tmp/doc-exec-9, inifile: 
+    collected 50 items
+    
+    test_50.py FF
+    
+    ================================= FAILURES =================================
+    _______________________________ test_num[17] _______________________________
+    
+    i = 17
+    
+        @pytest.mark.parametrize("i", range(50))
+        def test_num(i):
+            if i in (17, 25):
+    >          pytest.fail("bad luck")
+    E          Failed: bad luck
+    
+    test_50.py:6: Failed
+    _______________________________ test_num[25] _______________________________
+    
+    i = 25
+    
+        @pytest.mark.parametrize("i", range(50))
+        def test_num(i):
+            if i in (17, 25):
+    >          pytest.fail("bad luck")
+    E          Failed: bad luck
+    
+    test_50.py:6: Failed
+    ================= 2 failed, 48 deselected in 0.02 seconds ==================
+
+You have run only the two failing test from the last run, while 48 tests have
+not been run ("deselected").
+
+Now, if you run with the ``--ff`` option, all tests will be run but the first
+previous failures will be executed first (as can be seen from the series
+of ``FF`` and dots)::
+
+    $ py.test --ff
+    =========================== test session starts ============================
+    platform linux2 -- Python 2.7.6, pytest-2.7.3.dev426+ng9a90aac.d20150916, py-1.4.30, pluggy-0.3.0
+    run-last-failure: rerun last 2 failures first
+    rootdir: /tmp/doc-exec-9, inifile: 
+    collected 50 items
+    
+    test_50.py FF................................................
+    
+    ================================= FAILURES =================================
+    _______________________________ test_num[17] _______________________________
+    
+    i = 17
+    
+        @pytest.mark.parametrize("i", range(50))
+        def test_num(i):
+            if i in (17, 25):
+    >          pytest.fail("bad luck")
+    E          Failed: bad luck
+    
+    test_50.py:6: Failed
+    _______________________________ test_num[25] _______________________________
+    
+    i = 25
+    
+        @pytest.mark.parametrize("i", range(50))
+        def test_num(i):
+            if i in (17, 25):
+    >          pytest.fail("bad luck")
+    E          Failed: bad luck
+    
+    test_50.py:6: Failed
+    =================== 2 failed, 48 passed in 0.04 seconds ====================
+
+.. _`config.cache`:
+
+The new config.cache object
+--------------------------------
+
+.. regendoc:wipe
+
+Plugins or conftest.py support code can get a cached value
+using the pytest ``config`` object.  Here is a basic example
+plugin which implements a `funcarg <http://pytest.org/latest/funcargs.html>`_
+which re-uses previously created state across py.test invocations::
+
+    # content of test_caching.py
+    import time
+
+    def pytest_funcarg__mydata(request):
```

Comment:
I think it would be better to use the more modern `@pytest.fixture` decorator here